src/xpub.cpp

```diff
@@ -149,14 +149,20 @@ void zmq::xpub_t::xread_activated (pipe_t *pipe_)
             //  we can't give them back to userspace as it would be an API
             //  breakage since the payload of the message is completely
             //  different. Manually craft an old-style message instead.
-            data = data - 1;
-            size = size + 1;
+            //  Since with inproc we do not store the sub/cancel command
+            //  string anymore in the data buffer, there would be no room
+            //  to hack in the 0/1 byte, and it overwrites the metadata
+            //  pointer in the message struct, so create a new blob_t.
+            //  The pushback makes a copy of the data array anyway, so it
+            //  does not make any difference.
```

Comment:
I am not sure what this is meant to say and if it is still applicable. There is now the move below, as we discussed yesterday, so there is no copy during the push_back? I guess this also refers to the previous implementation, but this is very confusing in the code comment.

Reply:
Clarified a bit. It's already in the commit message, but it's a counter-intuitive place to be doing this processing so I think it's better to have inline comments too.

Reply:
Thanks for clarifying!